.github/workflows/gin.yml

```diff
@@ -0,0 +1,63 @@
+name: Run Tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        go: [1.12, 1.13, 1.14, 1.15]
+        test-tags: ['', nomsgpack]
+    name: ${{ matrix.os }} @ Go ${{ matrix.go }} ${{ matrix.test-tags }}
+    runs-on: ${{ matrix.os }}
+    env:
+      GO111MODULE: on
+      TESTTAGS: ${{ matrix.test-tags }}
+      GOPROXY: https://proxy.golang.org
+    steps:
+      - name: Set up Go ${{ matrix.go }}
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go }}
+
+      - name: Checkout Code
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.ref }}
+
+      - name: Install Dependencies
+        run: make tools
+
+      - name: Run Check
+        run: |
+          make vet
+          make fmt-check
+          make misspell-check
+
+      - name: Run Tests
+        run: make test
+
+      - name: Upload coverage to Codecov
+        if: success() && matrix.go == 1.15 && matrix.os == 'ubuntu-latest' && matrix.test-tags == 'nomsgpack'
```

Comment:
why `matrix.go == 1.15` here?

Reply:
If the `go test` is successful, the coverage is theoretically the same every time

So here we just need to avoid multiple uploads and choose one of them

Reply:
@laojianzi ok, thank your reply.

Reply:
1.12.x: https://travis-ci.org/github/gin-gonic/gin/jobs/753866331#L1150
1.13.x: https://travis-ci.org/github/gin-gonic/gin/jobs/753866332#L1150
...

@thinkerou In Travis CI though, it seems that the go test coverage is uploaded for every go version, so I can update it if you want.

Reply:
@laojianzi yes, please update, and remove `.travis.yml` and update gitter-room-id,thanks

Reply:
@thinkerou updated